internal/olm/operator/registry/index_image.go

```diff
@@ -146,8 +465,51 @@ func (c IndexImageCatalogCreator) UpdateCatalog(ctx context.Context, cs *v1alpha
 			// if no annotations exist and image reference is empty, error out
 			return errors.New("cannot upgrade: no catalog image reference exists in catalog source spec or annotations")
 		}
+
 		// if no annotations exist and image reference exists, set it to index image
 		c.IndexImage = cs.Spec.Image
+
+		// check if index image adopts File-Based Catalog or SQLite index image format
+		isFBCImage, err := isFBC(ctx, c.IndexImage)
+		if err != nil {
+			return err
+		}
+		c.HasFBCLabel = isFBCImage
+
+		// Upgrade a bundle that was installed using OLM
+		if c.HasFBCLabel {
+			// bundle add modes are not supported for FBC
+			if c.BundleAddMode != "" {
+				return fmt.Errorf("specifying the bundle add mode is not supported for File-Based Catalog bundles and index images")
+			}
+
+			// Upgrading when installed traditionally by OLM
+			upgradedFBC, err := handleTraditionalUpgrade(ctx, c.IndexImage, c.BundleImage, subscription.Spec.Channel)
+			if err != nil {
+				return err
+			}
+			c.FBCContent = upgradedFBC
+		}
+	} else {
+		// check if index image adopts File-Based Catalog or SQLite index image format
+		isFBCImage, err := isFBC(ctx, c.IndexImage)
+		if err != nil {
+			return err
+		}
+		c.HasFBCLabel = isFBCImage
+
+		// Upgrade an installed bundle from catalog source annotations
+		if c.HasFBCLabel {
+			// bundle add modes are not supported for FBC
+			if c.BundleAddMode != "" {
+				return fmt.Errorf("specifying the bundle add mode is not supported for File-Based Catalog bundles and index images")
+			}
+
+			err = setupFBCupdates(&c, ctx)
```

Comment:
As context should be the first parameter: 

```suggestion
			err = setupFBCupdates(ctx, &c)
```